velox/common/caching/CachedFactory.h

```diff
@@ -39,6 +39,16 @@
 
 namespace facebook::velox {
 
+struct Metadata {
```

Comment:
This needs to be a typename template in `CacheFactory` and only becomes concrete in `FileHandleCacheFactory`.  Also name it `FileProperies` there.

Reply:
Added new `typename Properties`